mypy/type_visitor.py

```diff
@@ -282,7 +282,7 @@ def visit_deleted_type(self, t: DeletedType) -> T:
         return self.strategy([])
 
     def visit_type_var(self, t: TypeVarType) -> T:
-        return self.strategy([])
+        return self.query_types([t.upper_bound] + t.values)
```

Comment:
This is strictly speaking unrelated, but it made one new test fail (because `has_praceholder()` was wrong).